terraform/staging.tfvars

```diff
@@ -30,6 +30,9 @@ website_managed_waf_rules = {
     metric_visibility = true
   }
 }
+website_feature_flags = {
```

Comment:
this ensures that feature-flags are similar to environment variables in that they are either on/off for a given environment, correct?

If that is the case it looks like we will lose the current functionality of being able to toggle the flag for any environment which is helpful when we want to support two pieces of functionality in parallel.

lmk if I'm misunderstanding the implementation.

Reply:
@as1729 The logic that "activates" a feature flag is entirely dependent on what the relevant FE code chooses to evaluate. The primary change in this PR is really just two things:
1. Establishing a nested `window.APP_CONFIG.featureFlags` object and moving all existing (currently there's only one) feature flags into that object, rather than having them set directly on `window.APP_CONFIG`.
2. Adding this `website_feature_flags` Terraform input variable and wiring it up to populate the `window.APP_CONFIG.featureFlags` object directly, instead of requiring multiple edits throughout the Terraform code in order to set feature flags.

The use of environment variables in evaluating the "status" of a flag is not interrupted (see [this line](https://github.com/usdigitalresponse/usdr-gost/pull/1754/files#diff-31ae1b1a4f152b03716b188dbd79cda77f6fb27e8a4e1dfeae8467f5b436cd0bR7)), although it's worth noting that env vars are only effective in dev environments – Vue/webpack only evaluate env vars at build-time (and the way they get built makes them immutable), which makes them unsuitable for dynamic toggling and/or setting different values between Staging and Production. You can think of it this way:
- In dev environments, use environment variables to set and source feature flags.
- In Staging or Prod environments, use `window.APP_CONFIG.featureFlags` to set and source feature flags.
    - It's worth noting that `window.APP_CONFIG.featureFlags` _could_ be used in dev environments, but we don't currently have any mechanism to populate this object in dev environments, so it would have to be done manually in the browser console.

This effectively means that there are two potential sources for a feature flag's value: an associated env var (e.g. `VUE_APP_USE_NEW_TABLE`) and a runtime variable (e.g. `window.APP_CONFIG.featureFlags.useNewTable`), and the effective value is dependent on how the FE code chooses to evaluate them. In the case of the (currently only) `useNewTable` feature flag, the FE code considers it "active" as long as either are set to `true`.

Eventually, I think it would be worth moving entirely to `window.APP_CONFIG.featureFlags` to remove this ambiguity and establish a single way to manage feature flags, but as mentioned above, we currently lack an established method for populating `window.APP_CONFIG.featureFlags` automatically in dev environments.

Reply:
Got it, makes sense, thanks!

Reply:
@as1729 The most-recent batch of updates to this branch contradicts a few of my statements in the above comment, namely that I added tooling to consolidate sourcing feature flags from `window.APP_CONFIG.featureFlags`, meaning that env vars are no longer needed. However, part of this update enables developers to modify feature flags in development environments by making changes to the (dev-only) `packages/client/public/deploy-config.js` file.

Let me know if that raises any concerns on your end.

Reply:
These changes look good. Agreed on eliminating need to set env-vars. I think it does simplify the implementation.